Technical Onboarding (Web Framework) - zirah

ziriraha opened the pull request.

ziriraha committed:
[ADD] awesome_owl: add a small view with a counter and a todolist.

This module contains a view with a simple todo list and a counter with also the components used to make them.

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/tutorials/pull/966.png)](https://mergebot.odoo.com/odoo/tutorials/pull/966)

ziriraha committed:
[ADD] awesome_dashboard: add a dashboard.

Using this dashboard a use can view information about their orders in real time (updated data).
The dashboard is customizable too, a user can decide which cards to see.

cgun-odoo commented on awesome_dashboard/static/src/dashboard/configurationdialog/configurationdialog.js:
```diff
@@ -0,0 +1,32 @@
+        this.items = useState(this.props.items.map((item) => {
+            return {
+                ...item,
+                enabled: !this.props.disabledItems.includes(item.id),
```
Just initially this looks like it might get confusing to store disabled items but keep the variable `enabled` on items instead of `disabled`. I'll see if it reduces any complexity though 🤔

ziriraha replied:
ok, ill make a change to make it a disabled variable instead.

cgun-odoo commented on awesome_dashboard/static/src/dashboard/configurationdialog/configurationdialog.js:
```diff
@@ -0,0 +1,32 @@
+        this.props.close();
+    }
+
+    onChange(checked, changedItem) {
```
You're getting checked but you're passing the function `ev`. Shouldn't you pass it `ev.target.checked` instead?

ziriraha replied:
You are right, I will rename the variable.

The Checkbox component has an onChange function:
```
onChange(ev) {
        if (!this.props.disabled) {
            this.props.onChange(ev.target.checked);
        }
    }
```
This function then calls the onChange you see in the configuration dialog file as it is passed as a prop.

cgun-odoo commented on awesome_dashboard/static/src/dashboard/dashboarditem/dashboarditem.js:
```diff
@@ -0,0 +1,20 @@
+/** @odoo-module **/
```
I think you don't need to have this comment anymore. For V17 it was actually necessary for odoo to interpret this file 👴. You can delete it and try if it still works

cgun-odoo commented on awesome_dashboard/static/src/dashboard/dashboarditem/dashboarditem.js:
```diff
@@ -0,0 +1,20 @@
+        },
+        size: {
+            type: Number,
+            default: 1,
```
default is not a keyword for props. If you want default values you should use `defaultProps`.  This is probably not working as you expected.

cgun-odoo commented on awesome_dashboard/static/src/dashboard/numbercard/numbercard.js:
```diff
@@ -0,0 +1,9 @@
+export class NumberCard extends Component {
+    static template = "awesome_dashboard.NumberCard";
+    static props = {
+        title: { type: String },
+        value: { type: Number },
```
```suggestion
        title: String,
        value: Number,
```
If you're only passing type no need to make it an object

cgun-odoo commented on awesome_dashboard/static/src/dashboard/numbercard/numbercard.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates xml:space="preserve">
+    <t t-name="awesome_dashboard.NumberCard" owl="1">
```
what is `owl="1"`?

cgun-odoo commented on awesome_dashboard/static/src/dashboard/piechart/piechart.js:
```diff
@@ -0,0 +1,40 @@
+        this.canvasRef = useRef("canvas");
+        onWillStart(() => loadJS(["/web/static/lib/Chart/Chart.js"]));
+        onMounted(() => this.renderChart());
+        onWillUnmount(() => this.chart.destroy());
```
This might give you an error if `this.chart` is undefined when you unmount the component. You can use `this.chart?.destroy();` instead.
The question mark checks if the left side is undefined and runs the function only if it isn't

cgun-odoo commented on awesome_dashboard/static/src/dashboard/piechart/piechart.js:
```diff
@@ -0,0 +1,40 @@
+        const color = labels.map((_, index) => getColor(index));
+
+        this.chart = new Chart(this.canvasRef.el, {
+            type: "doughnut",
```
Shouldn't it be a `Pie` chart like the name suggests 🥧

cgun-odoo commented on awesome_dashboard/static/src/dashboard/dashboard.js:
```diff
@@ -0,0 +1,57 @@
+
+    setup() {
+        this.action = useService("action");
+        this.stats = useState(useService("awesome_dashboard.statistics"));
```
thanks

ziriraha commented on awesome_dashboard/static/src/dashboard/dashboard.js:
```diff
@@ -0,0 +1,57 @@
+import { useService } from "@web/core/utils/hooks";
+import { DashboardItem } from "./dashboarditem/dashboarditem";
+import { ConfigurationDialog } from "./configurationdialog/configurationdialog";
+import { browser } from "@web/core/browser/browser";
```
are you just referring to the import browser or all those imports?

cgun-odoo commented on awesome_dashboard/static/src/dashboard_loader.js:
```diff
@@ -0,0 +1,13 @@
+    static template = xml`
+        <LazyComponent bundle="'awesome_dashboard.dashboard'" Component="'AwesomeDashboard'" props="props"/>
+    `;
+
```
```suggestion
```

cgun-odoo reviewed (commented):
💯

cgun-odoo commented:
perfect :goat:
